src/model/Surface.cpp

```diff
@@ -1440,16 +1440,101 @@ namespace model {
+    // 1 - Add all the subsurface rough openings together (so overlap areas dont count twice)
+    double Surface_Impl::totalAreaOfSubSurfaces() const {
+      double tol = 0.01;
+      // iterate over all sub surfaces
+      // make a map of sub surface/roughOpening vertices (flattened)
+      // iterate over map
+      // check each subsurface for overlap with parent and not overlap with sibling
+      // if either fails, replace roughOpening vertices with original vertices
+      // Add up all the openign areas
+
+      // Records the rough opening of each sub surface
+      std::map<SubSurface, Point3dVector> roughOpenings;
+      // Get the flattened parent surface vertices
+      Transformation parentToXY = Transformation::alignFace(this->vertices());
+      std::vector<Point3d> parentVertices = parentToXY.inverse() * this->vertices();
+      // Make sure the parent surface is oriented clockwise
+      auto norm = openstudio::getOutwardNormal(parentVertices);
+      if (norm && norm->z() > 0) {
+        std::reverse(parentVertices.begin(), parentVertices.end());
+      }
+
+      // Get the flattened sub surface vertices for windows
       for (const SubSurface& subSurface : this->subSurfaces()) {
         if (istringEqual(subSurface.subSurfaceType(), "FixedWindow") || istringEqual(subSurface.subSurfaceType(), "OperableWindow")) {
-          windowArea += subSurface.multiplier() * subSurface.netArea();
+          auto opening = parentToXY.inverse() * subSurface.roughOpeningVertices();
+          auto norm = openstudio::getOutwardNormal(opening);
+          if (norm && norm->z() > 0) {
+            std::reverse(opening.begin(), opening.end());
+          }
+          roughOpenings[subSurface] = opening;
         }
       }
 
-      double wwr = windowArea / grossArea;
+      // Check each rough opening against the parent surface, if any vertices are outside
+      // the parent surfave then revert back to the original vertices
+      for (const auto& opening : roughOpenings) {
+        const SubSurface& subSurface = opening.first;
+        std::vector<Point3d> vertices = opening.second;
 
-      return wwr;
+        if (!openstudio::polygonInPolygon(vertices, parentVertices, tol)) {
+          // One or more vertices not within the parent surface so replace with the original vertices
+          auto vertices = parentToXY * subSurface.vertices();
+          auto norm = openstudio::getOutwardNormal(vertices);
+          if (norm && norm->z() > 0) {
+            std::reverse(vertices.begin(), vertices.end());
+          }
+          roughOpenings[subSurface] = vertices;
+          break;
+        }
+      }
+
+      // Check subsurfaces for overlaps
+      // (if I knew how I'd order themn left to right in x then we could test adjacent ones only)
+      for (const auto& opening1 : roughOpenings) {
+        for (const auto& opening2 : roughOpenings) {
+          if (opening1 == opening2) continue;
```

Comment:
I'd like a place to put a breakpoint. If clang-format isn't throwing on the lack of braces, something is wrong with our rules.

```suggestion
          if (opening1 == opening2) {
            continue;
          }
```